bug(aria): Compile fails after upgrading Aria from 21.0.3 to 21.0.4

### Is this a regression?

- [x] Yes, this behavior used to work in the previous version

### The previous version in which this bug was not present was

21.0.3

### Description

When I upgraded from @angular/aria 21.0.3 to 21.0.4, I got following errors (hundreds of them) when I try to compile. It appears to be related to my use of the accordion package. Beyond that, I'm not sure what is going on here.

```
✘ [ERROR] NG3004: Unable to import symbol DeferredContentAware.
  The symbol is not exported from /path/to/repo/node_modules/@angular/aria/types/accordion.d.ts (module '@angular/aria/accordion'). [plugin angular-compiler]

    ../../../../../users/[username]/path/to/repo/src/app/subscriptions/preferences-page/preferences-page-ui.component.ngtypecheck.ts:0:0:
      0 │
        ╵ ^

  The symbol is declared here.

    node_modules/@angular/aria/types/_deferred-content-chunk.d.ts:7:0:
      7 │ declare class DeferredContentAware {
        ╵ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

### Reproduction


Honestly, I wouldn't know where to begin trying to reproduce this.

### Expected Behavior

It should compile.

### Actual Behavior

It does not compile, and I get the errors shown above.

### Environment

```
Angular CLI       : 21.0.4
Angular           : 21.0.6
Node.js           : 22.17.0
Package Manager   : yarn 1.22.22
Operating System  : darwin arm64

┌───────────────────────────────────┬───────────────────┬───────────────────┐
│ Package                           │ Installed Version │ Requested Version │
├───────────────────────────────────┼───────────────────┼───────────────────┤
│ @angular-devkit/core              │ 21.0.4            │ ^21.0.4           │
│ @angular-devkit/schematics        │ 21.0.4            │ ^21.0.4           │
│ @angular/animations               │ 21.0.6            │ 21.0.6            │
│ @angular/aria                     │ 21.0.4            │ 21.0.4            │
│ @angular/build                    │ 21.0.4            │ ^21.0.4           │
│ @angular/cdk                      │ 21.0.5            │ ^21.0.5           │
│ @angular/cli                      │ 21.0.4            │ ^21.0.4           │
│ @angular/common                   │ 21.0.6            │ 21.0.6            │
│ @angular/compiler                 │ 21.0.6            │ 21.0.6            │
│ @angular/compiler-cli             │ 21.0.6            │ 21.0.6            │
│ @angular/core                     │ 21.0.6            │ 21.0.6            │
│ @angular/forms                    │ 21.0.6            │ 21.0.6            │
│ @angular/language-service         │ 21.0.6            │ 21.0.6            │
│ @angular/localize                 │ 21.0.6            │ ^21.0.6           │
│ @angular/platform-browser         │ 21.0.6            │ 21.0.6            │
│ @angular/platform-browser-dynamic │ 21.0.6            │ 21.0.6            │
│ @angular/router                   │ 21.0.6            │ 21.0.6            │
│ rxjs                              │ 7.8.2             │ ^7.6.0            │
│ typescript                        │ 5.9.2             │ ^5.9.2            │
│ zone.js                           │ 0.15.1            │ ~0.15.0           │
└───────────────────────────────────┴───────────────────┴───────────────────┘
```

--- a/src/aria/accordion/index.ts
+++ b/src/aria/accordion/index.ts
@@ -6,11 +6,4 @@
  * found in the LICENSE file at https://angular.dev/license
  */
 
-export {AccordionPanel} from './accordion-panel';
-export {AccordionGroup} from './accordion-group';
-export {AccordionTrigger} from './accordion-trigger';
-export {AccordionContent} from './accordion-content';
-
-// This needs to be re-exported, because it's used by the accordion components.
-// See: https://github.com/angular/components/issues/30663.
-export {DeferredContent as ɵɵDeferredContent} from '../private';
+export * from './public-api';

--- a/src/aria/grid/index.ts
+++ b/src/aria/grid/index.ts
@@ -6,7 +6,4 @@
  * found in the LICENSE file at https://angular.dev/license
  */
 
-export {Grid} from './grid';
-export {GridCell} from './grid-cell';
-export {GridRow} from './grid-row';
-export {GridCellWidget} from './grid-cell-widget';
+export * from './public-api';

